### PR TITLE
Add clipboard-polyfill w/ git auto-update via single package.json

### PR DESCRIPTION
**Pull request by jbutz**
Pull request for issue: #11994

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/lgarron/clipboard-polyfill
 * Official website (optional, not the repository): https://garron.net/code/clipboard-polyfill/
 * NPM package url (optional): https://www.npmjs.com/package/clipboard-js
 * License and its reference: [MIT](https://github.com/lgarron/clipboard-polyfill/blob/master/LICENSE.md)
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 7
   - Count of stars: 231
   - Count of forks: 48
 * NPM download stats (optional):
   - Downloads in the last day: 1825
   - Downloads in the last week: 9386
   - Downloads in the last month: 44717

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [X] This lib was not found on cdnjs repo
 * [X] No already exist / duplicated issue and PR
 * [X] The lib has notable popularity
   * [X] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [X] More than 800 downloads stats per month on npm registry
 * [X] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [X] Auto-update setup
 * [ ] Auto-update target/source is valid.
 * [ ] Auto-update filemap is correct.

# Git commit checklist
 * [X] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [X] The parent of the commit(s) in the PR is not old than 3 days.
 * [X] Pull request is sending from a non-master branch with meaningful name.
 * [X] Separate unrelated changes into different commits.
 * [ ] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [ ] Close corresponding issue in commit message
 * [ ] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by jbutz** [comment]
Commits squashed 👍

**Comment by sufuf3** [comment]
hmmm...
As README said the install in npm is `npm install clipboard-polyfill` https://github.com/lgarron/clipboard-polyfill#usage
I'll help modify.

**Comment by PeterDaveHello** [comment]
So what's the current status here? @sufuf3 @maruilian11 
I think the PR title should also be revised?

File: ajax/libs/clipboard-polyfill/package.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    "cut",
+    "paste",
+    "execCommand",
+    "setData",
+    "getData",
+    "polyfill"
+  ],
+  "author": "Lucas Garron <code@garron.net> (https://garron.net/)",
+  "license": "MIT",
+  "npmName": "clipboard-js",
```
Comment (extend1994) [inline ajax/libs/clipboard-polyfill/package.json]: Let's use git auto-update instead. There are more complete versions on git repo.
```js
  "autoupdate": {
    "source": "git",
    "target": "git://github.com/lgarron/clipboard-polyfill.git",
    "fileMap": [
      {
        "basePath": "build",
        "files": [
          "**/!(*.ts)"
        ]
      },
      {
        "basePath": "",
        "files": [
          "clipboard*.js"
        ]
      }
    ]
  }
```

Use "build" as basePath to grab latest versions and use ""(root) as basePath to grab old versions of the lib.

**Comment by extend1994** [comment]
The npm package for this lib should be http://npmjs.com/package/clipboard-polyfill instead of https://www.npmjs.com/package/clipboard-js in the above PR comment.

**Comment by jbutz** [comment]
@extend1994 I've adjusted the file, I'll try and remember to squash the commits and change the commit message later. I don't have cdnjs checked out on the machine I am currently on.

**Comment by extend1994** [comment]
@jbutz Thanks for quick response! Just take your time :smiley:

**Comment by PeterDaveHello** [comment]
Let's see if we can get this merged earlier, I'll help rebase and squash in this case. @jbutz in case to slow down your whole system please remember to take a look at our document and enable git sparseCheckout on this repository, thanks!

**Comment by jbutz** [comment]
@PeterDaveHello I'm having a hard time rebasing my branch onto master, even with sparse checkout. I don't think my machine has the diskspace to do it. Any tips?

**Comment by sufuf3** [comment]
I am reviewing this PR, please give me more time.
Let's me help you rebase to the master branch.

(If you'd like to try, you can use `git fetch origin master:master` and `git rebase master` to rebase the master branch in `clipboard-polyfill` branch.)

**Comment by maruilian11** [comment]
@sufuf3 I think we should add `0.4.0` and `0.4.1` immediately, like we usually do.

**Comment by PeterDaveHello** [comment]
We can add them later, no need to block a PR by this reason unless the missing versions are that important!

**Comment by PeterDaveHello** [comment]
@cdnjs/library-reviewer don't know why didn't you ask for homepage info in the package.json but as we need to manually add two missing versions later, please also add the homepage info, thanks.

**Comment by maruilian11** [comment]
@PeterDaveHello Do you mean https://garron.net/code/clipboard-polyfill/? I thought we don't see a page just for demo as a homepage.

**Pull request merged**